src/ast/passes/semantic_analyser.cpp

```diff
@@ -817,6 +818,43 @@ void SemanticAnalyser::visit(Call &call)
 
     call.type = CreateInet(buffer_size);
   }
+  else if (call.func == "pton")
+  {
+    if (!check_varargs(call, 1, 2))
+      return;
+
+    auto arg = call.vargs->at(0);
+    auto af_type = AF_INET;
+    if (call.vargs->size() == 2)
+    {
+      check_arg(call, Type::integer, 0);
+      auto af_type_val = bpftrace_.get_int_literal(arg);
+      if (af_type_val.has_value())
+      {
+        af_type = *af_type_val;
+      }
+      arg = call.vargs->at(1);
+    }
```

Comment:
Is it possible to infer ipv4/ipv6 from the string instead of the optional param?